lib/puppet/pops/evaluator/access_operator.rb

```diff
@@ -605,12 +605,29 @@ def access_PHostClassType(o, scope, keys)
       # does not have a title. This should probably be deprecated.
       #
       result = keys.each_with_index.map do |c, i|
+        # PUP-6083 - Using Class[Foo] is deprecated since an arbitrary foo will trigger a "resource not found"
+        # This check is needed for two cases below.
+        #
+        strict_check = lambda do
```

Comment:
Why is a lambda used here? It enforces one extra assignment and scope creation in every iteration of this loop. A method with two parameters (name and index) method seems like a cleaner solution.